ENIGMAsystem/SHELL/Universal_System/bufferstruct.cpp

```diff
@@ -234,6 +234,17 @@ void buffer_fill(int buffer, unsigned offset, int type, variant value, unsigned
     }
   }
 }
+  
+void *buffer_get_address(int buffer) {
+  #ifdef DEBUG_MODE
+  if (buffer < 0 or size_t(buffer) >= enigma::buffers.size() or !enigma::buffers[buffer]) {
+    DEBUG_MESSAGE("Attempting to access non-existing buffer " + toString(buffer), MESSAGE_TYPE::M_USER_ERROR);
+    return nullptr;
+  }
+  #endif
+  enigma::BinaryBuffer *binbuff = enigma::buffers[buffer];
+  return reinterpret_cast<void *>(&binbuff->data[0]);
```

Comment:
If this is a vector, please favor `vec.data()` over `&vec[0]`.

Reply:
aye aye, captain!